hack/ovn-kind-cno.sh

```diff
@@ -26,38 +35,30 @@ if ! command -v kubectl; then
   exit 1
 fi
 
-# Detect IP to use as API server
-API_IP=$(ip -4 addr | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | grep -v "127.0.0.1" | head -n 1)
-if [ -z "$API_IP" ]; then
-  echo "Error detecting machine IP to use as API server"
-  exit 1
-fi
-
-sed -i "s/apiServerAddress.*/apiServerAddress: ${API_IP}/" kind.yaml
-
 # Ensure reachability to host via Docker network
 if ! sudo iptables -C DOCKER-USER -j ACCEPT > /dev/null 2>&1; then 
   sudo iptables -I DOCKER-USER -j ACCEPT
 fi
 
+ # create the config file
+  cat <<EOF > ${KIND_CONFIG}
+# config for 1 control plane node and 2 workers (necessary for conformance)
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  ipFamily: ${IP_FAMILY:-ipv4}
+  disableDefaultCNI: true
+  podSubnet: ${CLUSTER_CIDR:-10.244.0.0/16}
+  serviceSubnet: ${SERVICE_CIDR:-10.96.0.0/12}
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
+EOF
+
+
 # Create KIND cluster
-kind create cluster --name ovn --kubeconfig ${HOME}/admin.conf --image kindest/node:${K8S_VERSION} --config=./kind.yaml -v ${OVN_KIND_VERBOSITY}
```

Comment:
because the controller is running in a pod, is able to get the right configuration without having to pass the kubeconfig